TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs

```diff
@@ -23,29 +24,35 @@ public TypeCobolCustomLanguageServer(IRPCServer rpcServer, Queue<MessageActionWr
         /// </summary>
         public bool UseOutlineRefresh { get; set; }
 
-        protected override void OnShutdown()
-        {
-            if (UseOutlineRefresh && this.Workspace.DocumentModifiedEvent != null)
-                this.Workspace.DocumentModifiedEvent -= DocumentModified;
+        /// <summary>
+        /// Are we using the CFG view in the client.    
+        /// </summary>
+        public bool UseCfgDfaDataRefresh { get; set; }
 
-            base.OnShutdown();
+        protected override InitializeResult OnInitialize(InitializeParams parameters)
+        {
+            var result = base.OnInitialize(parameters);
+            this.Workspace.DocumentModifiedEvent += DocumentModified;
+            this.Workspace.ClientConfigurationChangedEvent += OnClientOptionsChanged;
+            return result;
         }
 
-        protected override void OnDidOpenTextDocument(DidOpenTextDocumentParams parameters)
+        /// <summary>
+        /// Handle -ol and -cfg options from the client configuration change notification.
+        /// </summary>
+        /// <param name="sender">Should be the Workspace instance</param>
+        /// <param name="options">Client's Options</param>
+        private void OnClientOptionsChanged(object sender, IEnumerable<string> options)
         {
-            if (UseOutlineRefresh && this.Workspace.DocumentModifiedEvent == null)
-            {
-                this.Workspace.DocumentModifiedEvent += DocumentModified;
-            }
-            base.OnDidOpenTextDocument(parameters);
+            //this.UseOutlineRefresh = options.Contains("-ol");
```

Comment:
Ok it is uncommented in  commit "WI #1818 Add LSR CfgDfaParams test c8f5286 ", because otherwise all tests that used -ol were in failure, but now the flags is not taken in account if the server is excuted in LSR test mode.